fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/PartitionPruner.java

```diff
@@ -134,39 +143,111 @@ public static <K> List<K> prune(List<Slot> partitionSlots, Expression partitionP
             return ImmutableList.of();
         }
 
-        List<OnePartitionEvaluator> evaluators = Lists.newArrayListWithCapacity(idToPartitions.size());
-        for (Entry<K, PartitionItem> kv : idToPartitions.entrySet()) {
-            evaluators.add(toPartitionEvaluator(
-                    kv.getKey(), kv.getValue(), partitionSlots, cascadesContext, expandThreshold));
+        if (sortedPartitionRanges != null) {
+            RangeSet<MultiColumnBound> predicateRanges = partitionPredicate.accept(
+                    new PartitionPredicateToRange(partitionSlots), null);
```

Comment:
dt<'2020-02-01' will compute a empty predicateRanges, left bound is closed and right bound is open. e.g.['2020-02-01','2020-02-01')